internal/parser.go

```diff
@@ -261,3 +262,93 @@ func ParseSCTE35(ctx context.Context, w io.Writer, f io.Reader, o Options) error
 
 	return jp.Error()
 }
+
+func FilterPids(ctx context.Context, w io.Writer, f io.Reader, o Options) error {
+	reader := bufio.NewReader(f)
+	_, err := packet.Sync(reader)
+	if err != nil {
+		return fmt.Errorf("syncing with reader %w", err)
+	}
+	pat, err := psi.ReadPAT(reader)
```

Comment:
ReadPAT is dangerous because it reads until it finds a PAT. That means that packets will not be seen of filtered if they arrive before the PAT.

We need some way of looking at all the packets and  then still reat the PAT once we find it.